### PR TITLE
Remove deprecated function jax.random.shuffle

### PR DESCRIPTION
**Pull request by jakevdp**
This was deprecated in v0.1.66 (#2925) and then apparently forgotten.

### CODE REVIEW

**Review by froystig — APPROVED** [review]

**Comment by jakevdp** [comment]
This has gone stale – I'm going to do a new PR to finish this deprecation.
